scripts/ingests/utils.py

```diff
@@ -1,10 +1,253 @@
+     name        bibcode                 doi                                                                                description
+    ------ ------------------- ------------------------ ----------------------------------------------------------------------------------------------------------------------------------------------------
+    Kirk19 2019ApJS..240...19K 10.3847/1538-4365/aaf6af Preliminary Trigonometric Parallaxes of 184 Late-T and Y Dwarfs and an Analysis of the Field Substellar Mass Function into the Planetary Mass Regime
+
+    >>> test = search_publication(db, name='Smith')
+    No matching publications for Smith, Trying Smit
+    No matching publications for Smit
+    Use add_publication() to add it to the database.
+
+    See Also
+    --------
+    add_publication: Function to add publications in the database
+
+    """
+
+    verboseprint = print if verbose else lambda *a, **k: None
+
+    # Make sure a search term is provided
+    if name is None and doi is None and bibcode is None:
+        print("Name, Bibcode, or DOI must be provided")
+        return
+
+    not_null_pub_filters = []
+    if name:
+        fuzzy_query_name = '%' + name + '%'
+        not_null_pub_filters.append(db.Publications.c.name.ilike(fuzzy_query_name))
+    if doi:
+        not_null_pub_filters.append(db.Publications.c.doi.ilike(doi))
+    if bibcode:
+        not_null_pub_filters.append(db.Publications.c.bibcode.ilike(bibcode))
+
+    if len(not_null_pub_filters) > 0:
+        pub_search_table = db.query(db.Publications).filter(or_(*not_null_pub_filters)).table()
+
+    n_pubs_found = len(pub_search_table)
+
+    if n_pubs_found == 1:
+        verboseprint(f'Found {n_pubs_found} matching publications for {name} or {doi} or {bibcode}')
+        if verbose:
+            pub_search_table.pprint_all()
+        return True
+
+    if n_pubs_found > 1:
+        print(f'Found {n_pubs_found} matching publications for {name} or {doi} or {bibcode}')
+        if verbose:
+            pub_search_table.pprint_all()
+        return pub_search_table
+
+    # If no matches found, search using first four characters of input name
+    if n_pubs_found == 0 and name:
+        shorter_name = name[:4]
+        print(f'No matching publications for {name}, Trying {shorter_name}')
+        fuzzy_query_shorter_name = '%' + shorter_name + '%'
+        pub_search_table = db.query(db.Publications).filter(db.Publications.c.name.ilike(fuzzy_query_shorter_name)).table()
+        n_pubs_found_short = len(pub_search_table)
+        if n_pubs_found_short == 0:
+            print(f'No matching publications for {shorter_name}')
+            print('Use add_publication() to add it to the database.')
+            return False
+
+        if n_pubs_found_short > 0:
+            print(f'Found {n_pubs_found_short} matching publications for {shorter_name}')
+            if verbose:
+                pub_search_table.pprint_all()
+            return pub_search_table
+
+    return
```

Comment:
Any thoughts on this final empty `return`?  I kinda just put it there to "close" the function but all the cases should have been covered above. @blackwer?